diff --git a/node_modules/file-selector/dist/file.d.ts b/node_modules/file-selector/dist/file.d.ts
index 9383de0..42dd248 100644
--- a/node_modules/file-selector/dist/file.d.ts
+++ b/node_modules/file-selector/dist/file.d.ts
@@ -1,5 +1,5 @@
 export declare const COMMON_MIME_TYPES: Map<string, string>;
 export declare function toFileWithPath(file: FileWithPath, path?: string): FileWithPath;
 export interface FileWithPath extends File {
-    readonly path?: string;
+    readonly path: string;
 }
